### PR TITLE
feat: removing some rules people dont like

### LINKED ISSUE #1132: Unrecommend openapi-tags
**User story.**
As a non-Stoplight user I do not understand the importance of tags, so it is a surprise when im told I absolutely must add them.

**Is your feature request related to a problem?**
Spectral users occasionally ask why Spectral is telling them to do things not in the OpenAPI spec. The answer is usually "this is a linter, not a validator" and that most of the rules are useful: avoiding bugs, stopping them getting hacked, etc. 

One rule which doesn't fit the description is `openapi-tags`. It _is_ useful for bigger APIs, but I'm working on a single endpoint OpenAPI description and adding tags to that is pretty redundant. It's there because Stoplight docs like tags, but for a single endpoint API I don't need my docs grouped like that anyhow.

**Describe the solution you'd like**

For v6.0 we should just make this one rule `recommended: false`.

**Additional context**

_In the future I'd like to look into a `spectral-stoplight-ruleset` with all sorts of rules relevant to Stoplight but not particularly relevant to other tools. Food for thought whilst working on this so we know the direction at least._

### LINKED ISSUE #1273: `operation-2xx-response` rule should not be recommended
I suggest to move `operation-2xx-response` rule out from recommended, as it is reasonable to have endpoints that have no success response but only 3xx redirects. Or maybe change the rule to allow 2xx or 3xx responses.

What do you think?

### PR DESCRIPTION
**Pull request by philsturgeon**
Fixes #1273
Fixes #1132

**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [x] Yes
- [ ] No

We're preparing for v6.0, which will include some major and excellent changes, but also some breaking changes to the bundled OpenAPI ruleset. These rules come from a bunch of feedback from the community about which rules should be showing up by default, and which are useful for "strict mode" (running with `extends: [[spectral:oas, all]]`. 

If you want any of these rules to stick around, speak now or forever hold your peace! 

**TLDR**

1. Bumping openapi-tag out to `recommended: false`(loads of people don't use tags and Studio doesn't help enough to make this easy for people who dont know what they are anyway). #1132
1. Changed operation-2xx-response to operation-success-response to solve #1273.
1. Removing operation-default-response altogether.

### CODE REVIEW

File: docs/reference/openapi-rules.md [hunk 1]
```diff
@@ -292,17 +288,6 @@ Use just one tag for an operation, which is helpful for some documentation syste
 
 **Recommended:** No
 
-### operation-summary-formatted
```
Comment (philsturgeon) [inline docs/reference/openapi-rules.md:295]: As we're prepping for v6 we can remove notes about v5.

File: docs/reference/openapi-rules.md [hunk 2]
```diff
@@ -253,10 +253,6 @@ Why? Well, you _can_ reference tags arbitrarily in operations, and definition is
 
 Defining tags allows you to add more information like a `description`. For more information see [tag-description](#tag-description).
 
-**Recommended:** Yes
-
-### operation-default-response
```
Comment (philsturgeon) [inline docs/reference/openapi-rules.md:258]: This makes very little sense to me, as the spec only says MAY have a default response.

File: .nvmrc [hunk 3]
```diff
@@ -1 +0,0 @@
-10.17.0
```
Comment (P0lip) [inline .nvmrc:1]: Could we bump it to 12 instead of removing it?
Me and nulltoken use it.

**Review by P0lip — COMMENTED** [review]
Looks good to me.

File: .nvmrc (hunk 3 above)
Comment (philsturgeon) [reply .nvmrc:1]: Done!

**Comment by philsturgeon** [comment]
Hey @lornajane you might like this one!

**Comment by philsturgeon** [comment]
@P0lip ready for thumbs!

**Review by P0lip — APPROVED** [review]

**Comment by P0lip** [comment]
Seems like we need to update test harness cases.

**Comment by P0lip** [comment]
@philsturgeon why did we merge this?
the tests weren't passing.

**Comment by philsturgeon** [comment]
I made a mistake, and thought it was azure being a dumbass and override it. We’ll get it working. I pushed a style fix to your test harness fixes to help get all the checks green. 

--
Phil Sturgeon
DevRel @ Stoplight.io

**Comment by dillonredding** [comment]
Any updates on this? I'd really like to get the `operation-success-response` rule 😁

**Comment by philsturgeon** [comment]
@dillonredding v6.0 is not yet released but if you'd like to go ahead and use this new rule you can always require the npm module via git and point it at the develop branch: https://stackoverflow.com/questions/39732397/install-specific-branch-from-github-using-npm
